packages/metro-serializer-esbuild/src/index.ts

```diff
@@ -44,15 +44,38 @@ function fixSourceMap(outputPath: string, text: string): string {
   return JSON.stringify({ ...sourcemap, sources });
 }
 
-function getSideEffects(modulePath: string): boolean | undefined {
-  const pkgJson = findPackage(modulePath);
-  if (!pkgJson) {
-    return undefined;
-  }
+const getSideEffects = (() => {
+  const pkgCache: Record<string, boolean | string[]> = {};
+  const getSideEffects = (pkgJson: string) => {
```

Comment:
```suggestion
  const getSideEffectsWorker = (pkgJson: string) => {
```
Differentiating these names makes it a little easier to mentally parse, and avoids potential scoping bugs.